bin/lib/cdn.py

```diff
@@ -0,0 +1,226 @@
+import hashlib
+import tarfile
+import urllib.parse
+
+from base64 import b64encode
+from concurrent import futures
+from datetime import datetime
+from tempfile import mkdtemp
+import os
+import shutil
+import logging
+from pathlib import Path
+
+from lib.amazon import botocore, s3_client, force_lazy_init
+
+logger = logging.getLogger('ce-cdn')
+
+def read_file_chunked(fobj, chunk_size=128*1024):
+    b = bytearray(chunk_size)
+    mv = memoryview(b)
+
+    for n in iter(lambda : fobj.readinto(mv), 0):
+        yield mv[:n]
+
+
+def hash_fileobj(fobj, hash_type):
+    h = hash_type()
+    for chnk in read_file_chunked(fobj):
+        h.update(chnk)
+    return h
+
+
+def hash_file_for_s3(f):
+    with open(f['path'], 'rb') as fobj:
+        sha256 = hash_fileobj(fobj, hashlib.sha256).digest()
```

Comment:
I think we could just use `.hexdigest()` here? It's a little longer than a base-64 encoded but at least agrees with the output of the `sha256sum` unix command